core/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java

```diff
@@ -135,6 +137,53 @@ public void testCreateShrinkIndexToN() {
         assertHitCount(client().prepareSearch("source").setSize(100).setQuery(new TermsQueryBuilder("foo", "bar")).get(), 20);
     }
 
+    public void testShrinkIndexPrimaryTerm() throws Exception {
+        final List<Integer> factors = Arrays.asList(2, 3, 5, 7);
+        final List<Integer> numberOfShardsFactors = randomSubsetOf(randomIntBetween(1, factors.size()), factors);
+        final int numberOfShards = numberOfShardsFactors.stream().reduce(1, (x, y) -> x * y);
+        final int numberOfTargetShards = randomSubsetOf(numberOfShardsFactors).stream().reduce(1, (x, y) -> x * y);
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        prepareCreate("source").setSettings(Settings.builder().put(indexSettings()).put("number_of_shards", numberOfShards)).get();
+
+        final ImmutableOpenMap<String, DiscoveryNode> dataNodes =
+                client().admin().cluster().prepareState().get().getState().nodes().getDataNodes();
+        assertThat(dataNodes.size(), greaterThanOrEqualTo(2));
+        final DiscoveryNode[] discoveryNodes = dataNodes.values().toArray(DiscoveryNode.class);
+        final String mergeNode = discoveryNodes[0].getName();
+        ensureGreen();
+
+        // restart random data nodes to force the primary term for some shards to increase
+        for (int i = 0; i < randomIntBetween(0, 16); i++) {
```

Comment:
why do we need up to 16 restarts? Maybe it's faster to fail shards by getting IndexShard instances from internalCluster? also this if loop calls `randomIntBetween(0, 16)` many times... I'm not sure that's what you intended